dataactvalidator/config/sqlrules/d3_detached_award_financial_assistance_1.sql

```diff
@@ -0,0 +1,8 @@
+-- Action type is required for non-aggregate records (i.e., when RecordType = 2)
+SELECT
+    row_number,
+    action_type,
+    record_type
+FROM detached_award_financial_assistance
+WHERE record_type = 2
+    AND (action_type = '' OR action_type IS NULL);
```

Comment:
I think if you don't enter anything into the column it automatically casts it to a null. Double-check that? It might alleviate your concerns about the query being complicated.

```
WHERE dafa.submission_id = {0}
    AND dafa.record_type = 2
    AND dafa.fain IS NULL
```
worked just fine for me in D1.